### PR TITLE
Fix `ref readonly` mismatch retargeting

### PR DESCRIPTION
**Pull request by jjonescz**
Follow up on https://github.com/dotnet/roslyn/pull/68680#discussion_r1267569309.

Without allowing mismatch the new test would fail for `emit = false`, i.e., PE and retargeting scenarios would be inconsistent.

Test plan: https://github.com/dotnet/roslyn/issues/68056

### CODE REVIEW

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1), assuming CI is passing

**Comment by jjonescz** [comment]
@jcouv for a second review, thanks

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Pull request merged**
